source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
Though technically this should work (host:port), it's possible that some webservers might not handle this correctly. I could go either way, but my recommendation for now would be to remove hostAndPortFromTcpUrl() function that you added, and just populate the hostname() with the host, and without the port. Happy to discuss though.